lib/rules/media-feature-name-value-allowed-list/index.js

```diff
@@ -37,70 +40,98 @@ const rule = (primary) => {
 		}
 
 		root.walkAtRules(/^media$/i, (atRule) => {
-			mediaParser(atRule.params).walk(/^media-feature-expression$/i, (node) => {
-				if (!node.nodes) return;
-
-				const mediaFeatureRangeContext = isRangeContextMediaFeature(node.parent.value);
-
-				// Ignore boolean
-				if (!node.value.includes(':') && !mediaFeatureRangeContext) {
-					return;
-				}
-
-				const mediaFeatureNode = node.nodes.find((n) => n.type === 'media-feature');
-
-				if (mediaFeatureNode == null) throw new Error('A `media-feature` node must be present');
-
-				let mediaFeatureName;
-				let values;
+			parseMediaQuery(atRule).forEach((mediaQuery) => {
+				if (isMediaQueryInvalid(mediaQuery)) return;
+
+				mediaQuery.walk(
+					(entry) => {
+						if (!entry.state) return;
+
+						if (isMediaFeature(entry.node)) {
+							entry.state.mediaFeatureName = entry.node.getName();
+							entry.state.unprefixedMediaFeatureName = vendor.unprefixed(entry.node.getName());
+
+							return;
+						}
+
+						if (!isMediaFeatureValue(entry.node)) return;
```

Comment:
question

Since the previous behaviour where it would `throw` if no media-feature was found wasn't tested, should we add a test or comment for your change?
Maybe it's too obvious and not necessary?

Reply:
I had interpreted the old behavior as a fixup for the typing of the old media query parser AST.

- `if (mediaFeatureNode == null) throw new Error('A `media-feature` node must be present');`
- `if (valueNode == null) throw new Error('A `value` node must be present');`

It doesn't make sense for either to ever be `null` in something that parses without error as a media query.

A boolean feature doesn't have a value, but there was check for that a few lines earlier.

-----

The new parser does't have this typing issue.